File: TI1/TI1.2 SOAP_DOTNET_SINBDD_GR09/02.CLICON/CONUNI_SOAP_DOTNET_CLICON_G09/CONUNI_SOAP_DOTNET_CLICON_G09/ec/edu/monster/modelo/Conversion.cs
using System;

namespace ec.edu.monster.modelo
{
    /// <summary>
    /// Modelo para representar una conversión
    /// Replicado desde Java - Patrón MVC
    /// </summary>
    public class Conversion
    {
        public string TipoConversion { get; set; }
        public string Operacion { get; set; }
        public double ValorOriginal { get; set; }
        public double ValorConvertido { get; set; }
        public string UnidadOriginal { get; set; }
        public string UnidadDestino { get; set; }
        public bool Exitosa { get; set; }
        public string MensajeError { get; set; }

        // Constructores
        public Conversion()
        {
        }

        public Conversion(string tipoConversion, string operacion, double valorOriginal, 
                         string unidadOriginal, string unidadDestino)
        {
            TipoConversion = tipoConversion;
            Operacion = operacion;
            ValorOriginal = valorOriginal;
            UnidadOriginal = unidadOriginal;
            UnidadDestino = unidadDestino;
            Exitosa = false;
        }

        public override string ToString()
        {
            if (Exitosa)
            {
                return $"{TipoConversion}: {ValorOriginal:F4} {UnidadOriginal} = {ValorConvertido:F4} {UnidadDestino}";
            }
            else
            {
                return $"Error en {TipoConversion}: {MensajeError}";
            }
        }
    }
}

File: Proyectos/TI.1 PAO202650_SINBDD_NRC30742/TI.1.3. PAO202650_RESTFUL_JAVA_SINBDD_NRC30742/04. CLICON/CONUNI_SOAP_JAVA_CLICON_G05/src/main/java/ec/edu/monster/modelo/Conversion.java
package ec.edu.monster.modelo;

/**
 * Modelo para representar una conversión
 * @author ACER NITRO V15
 */
public class Conversion {
    private String tipoConversion;
    private String operacion;
    private double valorOriginal;
    private double valorConvertido;
    private String unidadOriginal;
    private String unidadDestino;
    private boolean exitosa;
    private String mensajeError;

    // Constructores
    public Conversion() {
    }

    public Conversion(String tipoConversion, String operacion, double valorOriginal, 
                     String unidadOriginal, String unidadDestino) {
        this.tipoConversion = tipoConversion;
        this.operacion = operacion;
        this.valorOriginal = valorOriginal;
        this.unidadOriginal = unidadOriginal;
        this.unidadDestino = unidadDestino;
        this.exitosa = false;
    }

    // Getters y Setters
    public String getTipoConversion() {
        return tipoConversion;
    }

    public void setTipoConversion(String tipoConversion) {
        this.tipoConversion = tipoConversion;
    }

    public String getOperacion() {
        return operacion;
    }

    public void setOperacion(String operacion) {
        this.operacion = operacion;
    }

    public double getValorOriginal() {
        return valorOriginal;
    }

    public void setValorOriginal(double valorOriginal) {
        this.valorOriginal = valorOriginal;
    }

    public double getValorConvertido() {
        return valorConvertido;
    }

    public void setValorConvertido(double valorConvertido) {
        this.valorConvertido = valorConvertido;
    }

    public String getUnidadOriginal() {
        return unidadOriginal;
    }

    public void setUnidadOriginal(String unidadOriginal) {
        this.unidadOriginal = unidadOriginal;
    }

    public String getUnidadDestino() {
        return unidadDestino;
    }

    public void setUnidadDestino(String unidadDestino) {
        this.unidadDestino = unidadDestino;
    }

    public boolean isExitosa() {
        return exitosa;
    }

    public void setExitosa(boolean exitosa) {
        this.exitosa = exitosa;
    }

    public String getMensajeError() {
        return mensajeError;
    }

    public void setMensajeError(String mensajeError) {
        this.mensajeError = mensajeError;
    }

    @Override
    public String toString() {
        if (exitosa) {
            return String.format("%s: %.4f %s = %.4f %s", 
                tipoConversion, valorOriginal, unidadOriginal, 
                valorConvertido, unidadDestino);
        } else {
            return String.format("Error en %s: %s", tipoConversion, mensajeError);
        }
    }
}
